### PR TITLE
test: add sql runner e2e tests [postgres warehouse only]

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #11679 

## SQL Runner (new) Tests

1. Should verify that the query is autocompleted, run, and the results are displayed
   - Checks autocomplete functionality
   - Verifies query execution and result display
   - Confirms query saving in draft history
   - Tests query replacement with new table suggestion

2. Should verify that the chart is displayed
   - Checks initial disabled state of Run query button
   - Verifies query execution
   - Tests chart configuration and display
   - Validates adding new series to the chart
   - Checks grouping functionality
   - Verifies chart not displayed with incomplete configuration

3. Should verify that all chart types are displayed
   - Tests display of various chart types:
     - Table
     - Line chart
     - Pie chart
     - Bar chart

4. Should save a chart
   - Verifies chart saving process
   - Checks view mode after saving
   - Tests edit mode functionality
   - Verifies error handling and fixing in chart configuration
   - Confirms changes are reflected after saving and returning to view mode

5. Should not trigger an extra query to the warehouse when styling a chart
   - Verifies that styling changes don't trigger additional warehouse queries
   - Checks that chart updates correctly after styling changes

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by joaoviana** [comment]
image

**Review by rephus — APPROVED** [review]
Nice work, I added some comments, but they are not blocking.

File: packages/e2e/cypress/e2e/app/sqlRunnerNew.cy.ts [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+        // Verify that the table is displayed
+        cy.get(
+            `button[data-testid="visualization-${ChartKind.TABLE}"]`,
+        ).click();
+        cy.get('table thead th').eq(0).should('contain.text', 'customer_id');
+
+        // Verify that the line chart is displayed
+        cy.get(`button[data-testid="visualization-${ChartKind.LINE}"]`).click();
+        cy.get(`div[data-testid="chart-view-${ChartKind.LINE}"]`).should(
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/e2e/cypress/e2e/app/sqlRunnerNew.cy.ts:175]: if there's an error or we mess up the configuration generation we don't render the `Box` with the test id, so this is a safe test

File: packages/e2e/cypress/e2e/app/sqlRunnerNew.cy.ts [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+            'runPivotQuery',
+        );
+
+        cy.contains('Styling').click();
+        cy.contains('div', 'X-axis label')
+            .closest('.mantine-Stack-root')
+            .find('input.mantine-Input-input')
+            .type('{selectall}{backspace}New x-axis label');
+
+        cy.wait(2000).then(() => {
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/e2e/cypress/e2e/app/sqlRunnerNew.cy.ts]: Nice @rephus, I've improve this here! https://github.com/lightdash/lightdash/pull/11689/commits/350b752b760e2e02ac780df173a8152ee842ed6b

File: packages/e2e/cypress/e2e/app/sqlRunnerNew.cy.ts [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        cy.get('input[placeholder="Select group by"]').click();
+        cy.get('div[role="option"]').contains('first_name').click();
+        cy.get('.echarts-for-react')
+            .find('text')
+            .contains('Customer id aaron')
+            .should('be.visible');
+
+        // Verify that the chart is not displayed when the configuration is incomplete
+        cy.get('input[placeholder="Select X axis"]')
+            .siblings()
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/e2e/cypress/e2e/app/sqlRunnerNew.cy.ts:136]: Good point, improved the selector below that line: https://github.com/lightdash/lightdash/pull/11689/commits/255ff6ac32716fdca2602379165b3bd20241f490

**Pull request merged**